api.planx.uk/modules/auth/middleware.ts

```diff
@@ -109,3 +110,58 @@ export const useGoogleCallbackAuth: RequestHandler = (req, res, next) => {
     failureRedirect: "/auth/login/failed",
   })(req, res, next);
 };
+
+type UseRoleAuth = (authRoles: Role[]) => RequestHandler;
+
+/**
+ * Validate that an incoming request is using the role required for an endpoint
+ * Wrapped by the useJWT middleware to ensure token is valid, available, and decoded
+ *
+ * This does not check if a user can ultimately access a resource, only that they can access this route
+ * Hasura will validate this on a row and column basis when the query or mutation is made
+ */
+export const useRoleAuth: UseRoleAuth =
+  (authRoles) => async (req, res, next) => {
+    useJWT(req, res, () => {
```

Comment:
I spent a while trying to get these two middlewares a little less tightly coupled. The Express docs have a nice example of using an array of middleware - https://expressjs.com/en/guide/using-middleware.html

The issue I was hitting was keeping the return types for `req, res, next` outside this file when passing in the `role` argument to the middleware / function which returns a middleware array.

In reality they are tightly coupled - we can't check roles without decoding and validating the JWT, but I don't think this reads particularly clearly.